### PR TITLE
VZ-10925 Register CAPI clusters with Verrazzano when Rancher is not enabled/installed

### PR DESCRIPTION
**Pull request by jmaron99**
- Refactored CAPI cluster controller to process either Rancher or VZ registration
- Created VZ registration code
- Some modifications to existing VMC/MC code

### CODE REVIEW

File: cluster-operator/controllers/capi/rancher.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	if err != nil {
+		log.Errorf("Unable to unregister cluster %s from Rancher: %v", cluster.GetName(), err)
+		return err
+	}
+
+	return nil
+}
+
+// ensureRancherRegistration ensures that the CAPI cluster is registered with Rancher.
+func ensureRancherRegistration(ctx context.Context, r *RancherRegistration, cluster *unstructured.Unstructured) (ctrl.Result, error) {
```
Comment (mgianatagh) [inline cluster-operator/controllers/capi/rancher.go:120]: is this logic something that could be shared at some point with the existing sequence of steps for registering a cluster with Rancher?

File: cluster-operator/controllers/capi/verrazzano.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+		}
+	} else {
+		// need to create a CA secret in admin cluster
+		// get the workload cluster API CA cert
+		caCrt, err := v.getWorkloadClusterCACert(workloadClient)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		// persist the workload API certificate on the admin cluster
+		adminWorkloadCertSecret := &v1.Secret{ObjectMeta: metav1.ObjectMeta{
```
Comment (mgianatagh) [inline cluster-operator/controllers/capi/verrazzano.go:64]: do you know if we handle the expiration of this certificate?

**Review by mgianatagh — APPROVED** [review]

File: cluster-operator/controllers/capi/verrazzano.go (hunk 2 above)
Comment (jmaron99) [reply cluster-operator/controllers/capi/verrazzano.go:64]: Perhaps there's merit to continuously reconcile on relatively moderate cadence - that would pickup and update the cert secret if it changes

Comment (jmaron99) [reply cluster-operator/controllers/capi/verrazzano.go:64]: Apparently the VMC controller does handle workload cluster cert refresh

File: cluster-operator/controllers/capi/rancher.go (hunk 1 above)
Comment (mgianatagh) [reply cluster-operator/controllers/capi/rancher.go:120]: oh ok, I didn't realize it was already refactored

**Pull request merged**